File: iotonrails/app/models/tenant.rb

class Tenant < ApplicationRecord
  enum TenantType: [:Enterprise, :SMB, :ISV, :SystemIntegrator, :Government]


  has_many :Sites, class_name: 'Site'
  has_many :Users, class_name: 'TenantUser'
  has_many :Devices, class_name: 'IoTDevice'
  has_many :DataRetentionPolicies, class_name: 'DataRetentionPolicy'
  has_many :ConnectivityPlans, class_name: 'ConnectivityPlan'
  has_many :SimCards, class_name: 'SimCard'
  has_many :MessagingEndpoints, class_name: 'MessagingEndpoint'
  has_many :AccessPolicies, class_name: 'AccessPolicy'
  has_many :DeviceGroups, class_name: 'DeviceGroup'
  has_many :AlertRules, class_name: 'AlertRule'
  has_many :MaintenanceTickets, class_name: 'MaintenanceTicket'
  has_many :UsageRecords, class_name: 'UsageRecord'

end

File: iotOnGolang/internal/model/Tenant.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// Tenant Declaration
//==============================================================
type Tenant struct {
    gorm.Model
     Name                                    string
     Sites           []Site `gorm:"foreignKey:SitesFromTenantId"`
     Users           []TenantUser `gorm:"foreignKey:UsersFromTenantId"`
     Devices           []IoTDevice `gorm:"foreignKey:DevicesFromTenantId"`
     DataRetentionPolicies           []DataRetentionPolicy `gorm:"foreignKey:DataRetentionPoliciesFromTenantId"`
     ConnectivityPlans           []ConnectivityPlan `gorm:"foreignKey:ConnectivityPlansFromTenantId"`
     SimCards           []SimCard `gorm:"foreignKey:SimCardsFromTenantId"`
     MessagingEndpoints           []MessagingEndpoint `gorm:"foreignKey:MessagingEndpointsFromTenantId"`
     AccessPolicies           []AccessPolicy `gorm:"foreignKey:AccessPoliciesFromTenantId"`
     DeviceGroups           []DeviceGroup `gorm:"foreignKey:DeviceGroupsFromTenantId"`
     AlertRules           []AlertRule `gorm:"foreignKey:AlertRulesFromTenantId"`
     MaintenanceTickets           []MaintenanceTicket `gorm:"foreignKey:MaintenanceTicketsFromTenantId"`
     UsageRecords           []UsageRecord `gorm:"foreignKey:UsageRecordsFromTenantId"`
    TenantType                      TenantType

// parent associations as their child

}

